hatch/domain.py

```diff
@@ -0,0 +1,26 @@
+class Domain(object):
+
+    def __init__(self, subdomain, main_domain):
+        self.subdomain = subdomain
+        self.main_domain = main_domain
+
+    @property
+    def full_domain(self):
+        if self.subdomain:
+            return "{}.{}".format(self.subdomain, self.main_domain)
+        return self.main_domain
+
+    @staticmethod
+    def parse(domainStr):
+
+        if domainStr is None:
+            return None
+
+        parts = domainStr.split(".")
```

Comment:
This doesn't really work for some TLDs. 

Maybe just use https://github.com/john-kurkowski/tldextract